common/src/main/java/org/apache/falcon/entity/CatalogStorage.java

```diff
@@ -186,13 +193,14 @@ protected CatalogStorage(String uriTemplate) throws URISyntaxException {
         URI uri = new URI(processed);
 
         this.catalogUrl = uri.getScheme() + "://" + uri.getAuthority();
-
+        verifyAndUpdateConfiguration(getConf());
         parseUriTemplate(uri);
     }
 
     protected CatalogStorage(String uriTemplate, Configuration conf) throws URISyntaxException {
         this(uriTemplate);
         setConf(conf);
+        verifyAndUpdateConfiguration(conf);
```

Comment:
You don't need it here. verifyAndUpdateConfiguration is already called in CatalogStorage(String uriTemplate).